evalml/pipelines/classification_pipeline.py

```diff
@@ -54,6 +54,8 @@ def fit(self, X, y):
 
         Raises:
             ValueError: If the number of unique classes in y are not appropriate for the type of pipeline.
+            TypeError: If the dtype is boolean but pd.NA exists in the series.
```

Comment:
This isn't true, right? We catch that error and prevent it from being thrown.

Reply:
Yea, I'm not sure why I put that in there now